src/js/utils/validator.ts

```diff
@@ -85,7 +85,7 @@ export function isDeepKeyCompliant(defaultObj: GenericObject, customObj: Generic
             const customValue = customObj[key];
```

Comment:
does it not work if we do `const customValue: GenericObject = ...` here?

Reply:
Not exactly. It causes an `unsafe-any` on `customObj[key]` instead because it would return an `any` type.

But it could be changed to:
`const customValue = customObj[key] as GenericObject`

Reply:
oof

Reply:
typeof guards fix this:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types